ArjunaJTA/cdi/classes/com/arjuna/ats/jta/cdi/NarayanaTransactionManager.java

```diff
@@ -146,10 +155,13 @@
                              @Initialized(TransactionScoped.class)
                              final Event<Transaction> transactionScopeInitializedBroadcaster,
                              @Destroyed(TransactionScoped.class)
-                             final Event<Object> transactionScopeDestroyedBroadcaster) {
+                             final Event<Object> transactionScopeDestroyedBroadcaster,
```

Comment:
Hm... that's a good question. The old constructor does not exist.
I cannot imagine that somebody uses directly this constructor and that it can make an issue for backwards compatibility. But it could be safer to preserve the old one.

Reply:
Okay so you are saying that the only caller of the constructor should be this class.
What would be the impact of keeping the old constructor and calling this new one from it?

Reply:
That's not possible as the constructor is configured as an injection point for CDI.
The `@ApplicationScoped` transaction manager bean is constructed by CDI where CDI uses this constructor to create the bean. The `@Inject` above the constructor advices the CDI to inject all the constructor paramerers from the CDI runtime. It's CDI who needs to create the bean and injecting the parameters. We can't call it from the new one, it's CDI that construct the bean based on this constructor and there could be only one such constructor in a bean.

Reply:
The constructor is just a constructor and anything with package scope can call it - ie the fact that it's annotated with "@Inject" is irrelevant.

My point was that we are removing a method without deprecation and I was asking if you think there are no external callers.

Reply:
I see. Then yes, I think there are no external callers as the usage is with the CDI injection where CDI construct the bean based on the `@Inject` annotation. With that we can replace the constructor with this one.